benchmarks/poly.dx

```diff
@@ -0,0 +1,13 @@
+def poly {t} [VSpace t, Mul t] (x:t) : t =
+  4.0 .* (x * x * x * x) + 3.0 .* (x * x * x) + 2.0 .* (x * x) + x
```

Comment:
Perhaps use [`evalpoly`](https://github.com/google-research/dex-lang/blob/main/lib/prelude.dx#L2436)?

Reply:
We could also benchmark evalpoly vs the explicit method like this one.

Reply:
Done.  However, I note a slight infelicity: `evalpoly` requires an explicit `for` loop, because its argument `x` is typed as `Float`.  And we can't generalize the type in the desired way because we don't have a typeclass that would let us multiply objects of different types.

Reply:
Right, but that's a problem, because you've already fused the different operations. So it's technically already a significantly different program than the NumPy baseline.